### PR TITLE
feat: secure redis authentication [CVE-2024-31989]

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind bug

**What does this PR do / why we need it**:
Fix CVE-2024-31989

It has been discovered that an unprivileged pod in a different namespace on the same cluster could connect to the Redis server on port 6379. Despite having installed the latest version of the VPC CNI plugin on the EKS cluster, it requires manual enablement through configuration to enforce network policies. This raises concerns that many clients might unknowingly have open access to their Redis servers. This vulnerability could lead to Privilege Escalation to the level of cluster controller, or to information leakage, affecting anyone who does not have strict access controls on their Redis instance. This issue has been patched in version(s) 2.8.19, 2.9.15 and 2.10.10.

**Have you updated the necessary documentation?**
NA

**Which issue(s) this PR fixes**:

Fixes CVE-2024-31989

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: controllers/argocd/policyrule.go [hunk 1]
```diff
@@ -29,7 +29,22 @@ func policyRuleForApplicationController() []v1.PolicyRule {
...
+				"",
+			},
+			Resources: []string{
+				"secrets",
+			},
+			Verbs: []string{
+				"get",
+				"list",
+				"watch",
+				"create",
```
Comment (svghadi) [inline controllers/argocd/policyrule.go:44]: what do you think on limiting this only to get? If I am not wrong, Redis will only read the secret and creation will happen in operator manager.

Comment (iam-veeramalla) [reply controllers/argocd/policyrule.go]: Agreed. Made changes accordingly.

**Review by svghadi — APPROVED** [review]
Ran some manual test as well as ha kuttl test. LGTM. Thanks @iam-veeramalla

**Pull request merged**